app/src/main/java/de/markusfisch/android/wavelines/activity/EditorActivity.java

```diff
@@ -30,33 +24,48 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import de.markusfisch.android.wavelines.R;
+import de.markusfisch.android.wavelines.app.WaveLinesApp;
+import de.markusfisch.android.wavelines.database.Theme;
+import de.markusfisch.android.wavelines.widget.ThemeView;
+
 public class EditorActivity extends AppCompatActivity {
 	public static final String THEME_ID = "id";
 
 	private static final String THEME = "theme";
 
 	private final ArrayList<Integer> colors = new ArrayList<>();
-	private final SeekBar.OnSeekBarChangeListener updateColorFromBarsListener = new SeekBar.OnSeekBarChangeListener() {
-		@Override
-		public void onProgressChanged(SeekBar seekBar, int progressValue,
-				boolean fromUser) {
-			setColorFromHSVBars();
-			updateHSVLabels();
-			updatePreview();
-		}
-
-		@Override
-		public void onStartTrackingTouch(SeekBar seekBar) {
-		}
-
-		@Override
-		public void onStopTrackingTouch(SeekBar seekBar) {
-		}
-	};
+	private long themeId;
+	private ThemeView preview;
+	private SwitchCompat coupledSwitch;
+	private SwitchCompat uniformSwitch;
+	private SwitchCompat shuffleSwitch;
+	private TextView linesLabel;
+	private String linesTemplate;
+	private SeekBar linesBar;
+	private TextView wavesLabel;
+	private String wavesTemplate;
+	private SeekBar wavesBar;
+	private TextView amplitudeLabel;
+	private String amplitudeTemplate;
+	private SeekBar amplitudeBar;
+	private TextView oscillationLabel;
+	private String oscillationTemplate;
+	private SeekBar oscillationBar;
+	private TextView shiftLabel;
+	private String shiftLabelRandom;
+	private String shiftTemplate;
+	private SeekBar shiftBar;
+	private TextView speedLabel;
+	private String speedTemplate;
+	private SeekBar speedBar;
+	private TextView rotationLabel;
+	private String rotationTemplate;
+	private SeekBar rotationBar;
 	private final SeekBar.OnSeekBarChangeListener updateLabelsListener = new SeekBar.OnSeekBarChangeListener() {
 		@Override
 		public void onProgressChanged(SeekBar seekBar, int progressValue,
-				boolean fromUser) {
+									  boolean fromUser) {
```

Comment:
This is what I meant. IntelliJ indents with tabs until a tab wouldn't fit anymore (what makes no sense anyway because a tab is just a level of indent of unspecified width) and then use spaces for the rest 🙈

Reply:
lol yeah